src/main/play-doc/operation/ClusterUpgrade.md

```diff
@@ -0,0 +1,116 @@
+
+On a per-cluster basis, incoming requests are directed from the old cluster to the new cluster using load balancer membership or DNS changes.
+
+## Per node upgrade
+
+Per node upgrades are generally easier. However, they cannot be performed across ConductR releases that are marked as not compatible with previous releases.
+
+To perform a per node upgrade for ConductR Core, introduce new cluster members to the cluster. The new nodes could be created with updated versions of ConductR, Java, the Linux operating system or any other component installed during provisioning. As old members are removed from the cluster, bundles will be replicated to the new resources.
+
+To perform a per node upgrade for ConductR Agent, connect the new agent node to an existing cluster. As old agent nodes are removed from the cluster, bundles will be rescheduled for execution on the new agent nodes.
+
+To perform a per node upgrade for ConductR Core and ConductR Agent installed on the same host, connect both the ConductR Core and ConductR agent to an existing cluster. As old hosts (containing both old ConductR Core and ConductR Agent) are removed from the cluster, bundles will be replicated to the new resources, and bundles will be rescheduled for execution on the new agent nodes.
+
+It is critical to allow sufficient time for bundle relocation and replication before removing old members. In addition to ConductR replicating bundles to new nodes, stateful bundles may require additional time to replicate application data. Removing an agent node before application data has fully replicated can result in application data loss. Elasticsearch bundle provided along with ConductR, for example, requires [verification](#Elasticsearch-Verification) to ensure the data is transferred into the new agent node.
+
+Application specific data replication should also be monitored during an upgrade to prevent data loss.
+
+Be certain to ensure that sufficient resources for all roles are provisioned. Stopping the sole agent node providing a role leaves no nodes for bundles requiring that role to relocate to.
+
+## Per cluster upgrade
+
+To perform a per cluster upgrade, build a new cluster in isolation from the current running cluster. Once the new cluster is fully prepared, cut-over traffic using DNS, load balancers, routers, etc. Per cluster, upgrades may require more complicated strategies for migrating data storage managed by the cluster.
+
+## Elasticsearch Verification
+
+To perform deployment on a running cluster without downtime, Elasticsearch requires the new node to join the Elasticsearch cluster, the primary shard(s) is relocated from the old node to the new node, and Elasticsearch master is re-elected if required. Elasticsearch has the means to perform automatic relocation of primary shard(s) from the old node to the new node.
```

Comment:
State something like "this section only applies when ConductR manages Elasticsearch".